Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -189,6 +189,9 @@ class TIFFLoadingContext {
 
             auto const color_map = *m_metadata.color_map();
 
+            if (blue_offset + index >= color_map.size())
+                return Error::from_string_literal("TIFFImageDecoderPlugin: Color index is out of range");
```

Comment:
`index` is still an u16, so it could be 63, so that `1 * size` fits in a u64 (barely), `2 * size` overflows into `0`, this here isn't hit, and `red_offset` below is still out of bounds, right?

Reply:
Size doesn't barely fit in a `u64`, `m_metadata.bits_per_sample()` is at most 32 bits. So `blue_offset + index` can't take more than 34 bits.

Reply:
`m_metadata.bits_per_sample()` is used on the rhs of a right-shift operator. If it's 63, that's enough to fill the right-most bit in a 64-bit number. 63 fits in a 32-bit number. (It fits in a 6-bit number, even!)

…or an I misunderstanding?

Reply:
My bad, I wanted to say `m_metadata.bits_per_sample()` is at most 32, not 32 bits. We check the value here:

https://github.com/SerenityOS/serenity/blob/9507157f0412b2328b19190e2027417916b3ae64/Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp#L51-L52

Reply:
Ah, cool.